### ISSUE TITLE
iptables fails to start

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#5290]
**Describe the bug**
Getting the following in the logs:
```
Apr 03 14:34:58 test-fresh.inverse packetfence[21725]: WARN -e(21725): Problem trying to run command: LANG=C /sbin/iptables-restore < /usr/local/pf/var/conf/iptables.conf called from iptables_restore. Child exited with non-zero value 1 (pf::util::pf_run)
```

```
[root@localhost pf]# LANG=C /sbin/iptables-restore < /usr/local/pf/var/conf/iptables.conf
iptables-restore: line 205 failed
```

Line 205 of /usr/local/pf/var/conf/iptables.conf:
```
COMMIT
```

My full iptables.conf: https://support.inverse.ca/~jsemaan/iptables.conf

**Comment by fdurand** [comment]
NETFLOW kernel module is loaded ?

**Comment by julsemaan** [comment]
Doesn't look like it:
```
[root@localhost pf]# lsmod | grep -i netflow
(there is nothing)
```

**Comment by julsemaan** [comment]
Seems like the package is there:
```
[root@localhost pf]# rpm -qa | grep -i netflow
dkms-ipt-netflow-2.4-3.1.noarch
ipt-netflow-2.4-3.1.x86_64
```

**Comment by fdurand** [comment]
yum reinstall dkms-ipt-netflow --enablerepo=packetfence-devel

**Comment by julsemaan** [comment]
Looks like its the kernel headers issue again:
```
[root@localhost pf]# yum reinstall dkms-ipt-netflow --enablerepo=packetfence-devel
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.calgah.com
 * extras: centos.mirror.colo-serv.net
 * updates: centos.mirror.iweb.ca
Resolving Dependencies
--> Running transaction check
---> Package dkms-ipt-netflow.noarch 0:2.4-3.1 will be reinstalled
--> Finished Dependency Resolution

Dependencies Resolved

===========================================================================================================================================================================================================================================================================================================================================================================
 Package                                                                                      Arch                                                                               Version                                                                               Repository                                                                                     Size
===========================================================================================================================================================================================================================================================================================================================================================================
Reinstalling:
 dkms-ipt-netflow                                                                             noarch                                                                             2.4-3.1                                                                               packetfence-devel                                                                              58 k

Transaction Summary
===========================================================================================================================================================================================================================================================================================================================================================================
Reinstall  1 Package

Total download size: 58 k
Installed size: 212 k
Is this ok [y/d/N]: y
Downloading packages:
dkms-ipt-netflow-2.4-3.1.noarch.rpm                                                                                                                                                                                                                                                                                                                 |  58 kB  00:00:00     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : dkms-ipt-netflow-2.4-3.1.noarch                                                                                                                                                                                                                                                                                                                         1/1 
Error! DKMS tree already contains: ipt-netflow-2.4-3.1
You cannot add the same module/version combo more than once.
Error! echo
Your kernel headers for kernel 3.10.0-1062.4.1.el7.x86_64 cannot be found at
/lib/modules/3.10.0-1062.4.1.el7.x86_64/build or /lib/modules/3.10.0-1062.4.1.el7.x86_64/source.
Error! echo
Your kernel headers for kernel 3.10.0-1062.4.1.el7.x86_64 cannot be found at
/lib/modules/3.10.0-1062.4.1.el7.x86_64/build or /lib/modules/3.10.0-1062.4.1.el7.x86_64/source.
warning: %post(dkms-ipt-netflow-2.4-3.1.noarch) scriptlet failed, exit status 1
Non-fatal POSTIN scriptlet failure in rpm package dkms-ipt-netflow-2.4-3.1.noarch
  Verifying  : dkms-ipt-netflow-2.4-3.1.noarch                                                                                                                                                                                                                                                                                                                         1/1 

Installed:
  dkms-ipt-netflow.noarch 0:2.4-3.1                                                                                                                                                                                                                                                                                                                                        

Complete!
You have new mail in /var/spool/mail/root
```

**Comment by fdurand** [comment]
yes and didn'tfind a way to install the kernel headers that match your running kernel.
So you need to install the latest kernel and retry.

**Comment by julsemaan** [comment]
we need to have the proper procedure documented in our doc to state that before installing PacketFence, you need to have the latest kernel installed

**Comment by fdurand** [comment]
I can add in the doc something like : yum install "kernel-devel-uname-r == $(uname -r)"

**Comment by extrafu** [comment]
I think that's enough and should be part of our documentation.

**Comment by nqb** [comment]
Same issue occured for Debian 9 and documentation instructions are valid only for EL.

**Comment by nqb** [comment]
On Debian, it seems necessary to run:
```
apt install linux-headers-$(uname -r)
```

I'm running latest version of kernel headers for Debian (`4.9.210-1`) with `iptables-netflow-dkms` package installed from packetfence repo and still have the issue.

During PacketFence installation on Debian, I noticed following error message:
```
Setting up iptables-netflow-dkms (2.4-1) ...                                                                                                                                                                      
Loading new ipt-netflow-2.4 DKMS files...                                                                                                                                                                         
Building for 4.9.0-12-amd64                                                                                                                                                                                       
Building initial module for 4.9.0-12-amd64                                                                                                                                                                        
Error! Bad return status for module build on kernel: 4.9.0-12-amd64 (x86_64)                                                                                                                                      
Consult /var/lib/dkms/ipt-netflow/2.4/build/make.log for more information. 
```
but it doesn't stop installation.

Log file:
```
DKMS make.log for ipt-netflow-2.4 for kernel 4.9.0-12-amd64 (x86_64)
Tue Apr  7 06:03:12 UTC 2020
make: *** No rule to make target 'gen_compat_def', needed by 'compat_def.h'.  Stop.
```

**Comment by nqb** [comment]
Nothing change if I run: `dpkg-reconfigure iptables-netflow-dkms`.

**Comment by nqb** [comment]
We also need to change installation instructions to reflect this change.

**Comment by nqb** [comment]
@fdurand, could you upload new package for iptables-netflow-dkms on inverse.ca ?

I will take care of changes in docs.

**Comment by nqb** [comment]
I did the upload.

**Comment by nqb** [comment]
@fdurand, 

From what I see for Debian, we could add [linux-headers-amd64](https://packages.debian.org/stretch/linux-headers-amd64) (meta-package) as an installation dependency for PF.

**Comment by nqb** [comment]
> From what I see for Debian, we could add [linux-headers-amd64](https://packages.debian.org/stretch/linux-headers-amd64) (meta-package) as an installation dependency for PF.

It looks like it's already the case. @fdurand, if it's ok, let me know and I will remove Debian part from my PR.

**Comment by nqb** [comment]
`yum install kernel-devel` works on my side.

**Comment by fdurand** [comment]
Yes probably because you already have the latest kernel running.

**Comment by nqb** [comment]
As seen with @fdurand, on EL, if you run:
```
yum makecache
yum install kernel-devel
```
on an older CentOS 7 OS (not running latest kernel), `yum` will install latest `kernel-devel` that didn't match your current kernel version.

**Comment by rcmcdonald91** [comment]
I'm seeing this issue on fully patched 10.1.0 cluster. 

Ran this prior to installing PF

`yum install kernel-devel-$(uname -r)`



`[root@pf1 ~]# uname -a
Linux pf1.ad.mydomain.com 3.10.0-1127.13.1.el7.x86_64 #1 SMP Tue Jun 23 15:46:38 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`



> [root@pf1 ~]# systemctl status packetfence-iptables -l
> ● packetfence-iptables.service - PacketFence Iptables configuration
>    Loaded: loaded (/usr/lib/systemd/system/packetfence-iptables.service; enabled; vendor preset: disabled)
>    Active: active (running) since Tue 2020-07-07 21:06:38 EDT; 2min 40s ago
>  Main PID: 2482 (perl)
>    CGroup: /packetfence.slice/packetfence-iptables.service
>            └─2482 /bin/perl -I/usr/local/pf/lib -Mpf::db -Mpf::services::manager::iptables -e my $db ; while(!$db) { eval { $db = db_connect() } ; sleep 1 } ; pf::services::manager::iptables->new()->startAndCheck()
> 
> Jul 07 21:08:49 pf1.ad.mydomain.com sudo[3742]: pam_systemd(sudo:session): Failed to create session: Unit is masked.
> Jul 07 21:08:49 pf1.ad.mydomain.com sudo[3742]: pam_unix(sudo:session): session opened for user root by (uid=0)
> Jul 07 21:08:49 pf1.ad.mydomain.com sudo[3745]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/sbin/ipset --add pfsession_passthrough 172.217.13.99,443
> Jul 07 21:08:49 pf1.ad.mydomain.com sudo[3745]: pam_systemd(sudo:session): Failed to create session: Unit is masked.
> Jul 07 21:08:49 pf1.ad.mydomain.com sudo[3745]: pam_unix(sudo:session): session opened for user root by (uid=0)
> Jul 07 21:08:49 pf1.ad.mydomain.com sudo[3745]: pam_unix(sudo:session): session closed for user root
> Jul 07 21:08:49 pf1.ad.mydomain.com packetfence[2482]: -e(2482) INFO: Adding NAT Masquerade statement. (pf::iptables::generate_passthrough_rules)
> Jul 07 21:08:49 pf1.ad.mydomain.com packetfence[2482]: -e(2482) INFO: restoring iptables from /usr/local/pf/var/conf/iptables.conf (pf::iptables::iptables_restore)
> Jul 07 21:08:49 pf1.ad.mydomain.com perl[2482]: iptables-restore: line 205 failed
> Jul 07 21:08:49 pf1.ad.mydomain.com packetfence[2482]: -e(2482) WARN: Problem trying to run command: LANG=C /sbin/iptables-restore < /usr/local/pf/var/conf/iptables.conf called from iptables_restore. Child exited with non-zero value 1 (pf::util::pf_run)
